bufit.go

```diff
@@ -131,6 +131,9 @@ func (b *Buffer) NextReader() io.ReadCloser {
 // Write appends the given data to the buffer. All active readers will
 // see this write.
 func (b *Buffer) Write(p []byte) (int, error) {
+	if !b.alive() {
+		return 0, io.ErrClosedPipe
+	}
```

Comment:
Okay that makes sense. Is there any harm in the `b.cond.Broadcast()` happening in the error case?